PAT panics on template error in alert's definition

Hello,

Thank you for your software, it's pretty useful!

I'm trying to run following test file:

```yaml
name: Test template error
rules:
  fromLiteral: |-
    groups:
    - name: prometheus.rules
      rules:
      - alert: ClousSQLDiskUtilizationHigh
        expr: stackdriver_cloudsql_database_cloudsql_googleapis_com_database_disk_utilization * 100 > 80
        for: 1m
        labels:
          severity: '{{ if lt $value 95 }}warning{{ else }}critical{{ end }}'
          description: 'ClousSQL database {{ $labels.database_id }} disk utilization high: {{ $value }}% used'

fixtures:
- duration: 5m
  metrics:
  - stackdriver_cloudsql_database_cloudsql_googleapis_com_database_disk_utilization{database_id="test"} 0.853

assertions:
- at: 0m
  expected:
  - alertname: ClousSQLDiskUtilizationHigh
    database_id: test
    alertstate: pending
    severity: warning
    description: 'ClousSQL database test disk utilization high: 85.3% used'
- at: 5m
  expected:
  - alertname: ClousSQLDiskUtilizationHigh
    database_id: test
    alertstate: firing
    severity: warning
    description: 'ClousSQL database test disk utilization high: 85.3% used'
```

And end up with following trace:

```plain
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x11c98b0]

goroutine 1 [running]:
github.com/kevinjqiu/pat/vendor/github.com/go-kit/kit/log.(*context).Log(0xc4211f4fc0, 0xc4200925a0, 0x6, 0x6, 0x1, 0xc420068000)
	/Users/medvedchikov/projects/src/github.com/kevinjqiu/pat/vendor/github.com/go-kit/kit/log/log.go:124 +0x1a0
github.com/kevinjqiu/pat/vendor/github.com/prometheus/prometheus/rules.(*AlertingRule).Eval.func1(0xc4200ae140, 0x37, 0xb, 0xc4200ae280)
	/Users/medvedchikov/projects/src/github.com/kevinjqiu/pat/vendor/github.com/prometheus/prometheus/rules/alerting.go:220 +0x551
github.com/kevinjqiu/pat/vendor/github.com/prometheus/prometheus/rules.(*AlertingRule).Eval(0xc4200f4000, 0x160f9e0, 0xc420160980, 0x0, 0xe7791f700, 0x18bc8e0, 0xc420278400, 0x0, 0x0, 0x0, ...)
	/Users/medvedchikov/projects/src/github.com/kevinjqiu/pat/vendor/github.com/prometheus/prometheus/rules/alerting.go:228 +0x5c8
github.com/kevinjqiu/pat/pkg.PromRuleTest.evalRuleGroupAtInstant(0xc4200bd280, 0x13, 0x0, 0x0, 0xc42025e000, 0x182, 0xc4201045d0, 0x1d, 0xc4201b4ae0, 0x1, ...)
	/Users/medvedchikov/projects/src/github.com/kevinjqiu/pat/pkg/ruletest.go:36 +0x328
github.com/kevinjqiu/pat/pkg.PromRuleTest.GenerateTestCases(0xc4200bd280, 0x13, 0x0, 0x0, 0xc42025e000, 0x182, 0xc4201045d0, 0x1d, 0xc4201b4ae0, 0x1, ...)
	/Users/medvedchikov/projects/src/github.com/kevinjqiu/pat/pkg/ruletest.go:68 +0x2d7
github.com/kevinjqiu/pat/cmd.run(0xc4201f62e0, 0x1, 0x1, 0x0, 0x0)
	/Users/medvedchikov/projects/src/github.com/kevinjqiu/pat/cmd/root.go:56 +0x38f
github.com/kevinjqiu/pat/cmd.glob..func1(0x18b47e0, 0xc4201f62e0, 0x1, 0x1)
	/Users/medvedchikov/projects/src/github.com/kevinjqiu/pat/cmd/root.go:80 +0x3f
github.com/kevinjqiu/pat/vendor/github.com/spf13/cobra.(*Command).execute(0x18b47e0, 0xc4200c2010, 0x1, 0x1, 0x18b47e0, 0xc4200c2010)
	/Users/medvedchikov/projects/src/github.com/kevinjqiu/pat/vendor/github.com/spf13/cobra/command.go:766 +0x2c1
github.com/kevinjqiu/pat/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x18b47e0, 0x1012728, 0x10, 0x151d020)
	/Users/medvedchikov/projects/src/github.com/kevinjqiu/pat/vendor/github.com/spf13/cobra/command.go:852 +0x30a
github.com/kevinjqiu/pat/vendor/github.com/spf13/cobra.(*Command).Execute(0x18b47e0, 0xc42009c058, 0x0)
	/Users/medvedchikov/projects/src/github.com/kevinjqiu/pat/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/kevinjqiu/pat/cmd.Execute()
	/Users/medvedchikov/projects/src/github.com/kevinjqiu/pat/cmd/root.go:88 +0x2d
main.main()
	/Users/medvedchikov/projects/src/github.com/kevinjqiu/pat/main.go:6 +0x20
```

I'm expecting some message like "you have an error in your severity field" instead of panic.

Go: go version go1.10.3 darwin/amd64
OS: Darwin Kernel Version 17.6.0
Revision used: 1cf3c774aecb078ed253ca15342c9b5d6630ccdc

Thank you.

@alexey-medvedchikov With the fix in #7, the output now looks like this:

```
./pat issue-6.yaml 
level=warn msg="Expanding alert template failed" err="error executing template __alert_ClousSQLDiskUtilizationHigh: template: __alert_ClousSQLDiskUtilizationHigh:1:48: executing \"__alert_ClousSQLDiskUtilizationHigh\" at <lt $value 95>: error calling lt: incompatible types for comparison" data="unsupported value type"
level=warn msg="Expanding alert template failed" err="error executing template __alert_ClousSQLDiskUtilizationHigh: template: __alert_ClousSQLDiskUtilizationHigh:1:48: executing \"__alert_ClousSQLDiskUtilizationHigh\" at <lt $value 95>: error calling lt: incompatible types for comparison" data="unsupported value type"
=== RUN   Test_template_error_at_0m
--- FAIL: Test_template_error_at_0m (0.00s)
	ruletest.go:212: 
			Error Trace:	ruletest.go:212
			            				ruletest.go:92
			Error:      	Not equal: 
			            	expected: []map[string]string{map[string]string{"description":"ClousSQL database test disk utilization high: 85.3% used", "__name__":"ALERTS", "alertname":"ClousSQLDiskUtilizationHigh", "database_id":"test", "alertstate":"pending", "severity":"warning"}}
			            	actual  : []map[string]string{map[string]string{"__name__":"ALERTS", "alertname":"ClousSQLDiskUtilizationHigh", "alertstate":"pending", "database_id":"test", "description":"ClousSQL database test disk utilization high: 85.3% used", "severity":"<error expanding template: error executing template __alert_ClousSQLDiskUtilizationHigh: template: __alert_ClousSQLDiskUtilizationHigh:1:48: executing \"__alert_ClousSQLDiskUtilizationHigh\" at <lt $value 95>: error calling lt: incompatible types for comparison>"}}
			            	
			            	Diff:
			            	--- Expected
			            	+++ Actual
			            	@@ -7,3 +7,3 @@
			            	   (string) (len=11) "description": (string) (len=56) "ClousSQL database test disk utilization high: 85.3% used",
			            	-  (string) (len=8) "severity": (string) (len=7) "warning"
			            	+  (string) (len=8) "severity": (string) (len=260) "<error expanding template: error executing template __alert_ClousSQLDiskUtilizationHigh: template: __alert_ClousSQLDiskUtilizationHigh:1:48: executing \"__alert_ClousSQLDiskUtilizationHigh\" at <lt $value 95>: error calling lt: incompatible types for comparison>"
			            	  }
			Test:       	Test_template_error_at_0m
=== RUN   Test_template_error_at_5m
--- FAIL: Test_template_error_at_5m (0.00s)
	ruletest.go:212: 
			Error Trace:	ruletest.go:212
			            				ruletest.go:92
			Error:      	Not equal: 
			            	expected: []map[string]string{map[string]string{"alertname":"ClousSQLDiskUtilizationHigh", "database_id":"test", "alertstate":"firing", "severity":"warning", "description":"ClousSQL database test disk utilization high: 85.3% used", "__name__":"ALERTS"}}
			            	actual  : []map[string]string{map[string]string{"database_id":"test", "description":"ClousSQL database test disk utilization high: 85.3% used", "severity":"<error expanding template: error executing template __alert_ClousSQLDiskUtilizationHigh: template: __alert_ClousSQLDiskUtilizationHigh:1:48: executing \"__alert_ClousSQLDiskUtilizationHigh\" at <lt $value 95>: error calling lt: incompatible types for comparison>", "__name__":"ALERTS", "alertname":"ClousSQLDiskUtilizationHigh", "alertstate":"firing"}}
			            	
			            	Diff:
			            	--- Expected
			            	+++ Actual
			            	@@ -7,3 +7,3 @@
			            	   (string) (len=11) "description": (string) (len=56) "ClousSQL database test disk utilization high: 85.3% used",
			            	-  (string) (len=8) "severity": (string) (len=7) "warning"
			            	+  (string) (len=8) "severity": (string) (len=260) "<error expanding template: error executing template __alert_ClousSQLDiskUtilizationHigh: template: __alert_ClousSQLDiskUtilizationHigh:1:48: executing \"__alert_ClousSQLDiskUtilizationHigh\" at <lt $value 95>: error calling lt: incompatible types for comparison>"
			            	  }
			Test:       	Test_template_error_at_5m
FAIL
```

--- a/pkg/bindata.go
+++ b/pkg/bindata.go
@@ -82,7 +82,7 @@ func schemaSchemaJson() (*asset, error) {
 		return nil, err
 	}
 
-	info := bindataFileInfo{name: "schema/schema.json", size: 1992, mode: os.FileMode(420), modTime: time.Unix(1529515698, 0)}
+	info := bindataFileInfo{name: "schema/schema.json", size: 1992, mode: os.FileMode(420), modTime: time.Unix(1530413875, 0)}
 	a := &asset{bytes: bytes, info: info}
 	return a, nil
 }
@@ -102,7 +102,7 @@ func schemaSchemaYaml() (*asset, error) {
 		return nil, err
 	}
 
-	info := bindataFileInfo{name: "schema/schema.yaml", size: 1294, mode: os.FileMode(420), modTime: time.Unix(1529515679, 0)}
+	info := bindataFileInfo{name: "schema/schema.yaml", size: 1294, mode: os.FileMode(420), modTime: time.Unix(1529521862, 0)}
 	a := &asset{bytes: bytes, info: info}
 	return a, nil
 }

--- a/pkg/ruleloader.go
+++ b/pkg/ruleloader.go
@@ -8,6 +8,8 @@ import (
 	"github.com/prometheus/prometheus/promql"
 	"time"
 	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/go-kit/kit/log"
+	"os"
 )
 
 func (ruleLoader RuleLoader) getRuleFilePath() string {
@@ -20,9 +22,12 @@ func (ruleLoader RuleLoader) getRuleFilePath() string {
 func (ruleLoader RuleLoader) Load() ([]*rules.Group, error) {
 	var (
 		ruleGroups *rulefmt.RuleGroups
-		errs []error
-		filename string
+		errs       []error
+		filename   string
+		logger     log.Logger
 	)
+
+	logger = log.NewLogfmtLogger(log.NewSyncWriter(os.Stdout))
 
 	switch {
 	case ruleLoader.FromFile != "":
@@ -60,7 +65,7 @@ func (ruleLoader RuleLoader) Load() ([]*rules.Group, error) {
 					time.Duration(r.For),
 					labels.FromMap(r.Labels),
 					labels.FromMap(r.Annotations),
-					nil,    // TODO: stub logger
+					logger,
 				)
 				rls = append(rls, rule)
 			}

